### ISSUE TITLE
Consider adding v2a_ref, v3a_ref to wcsinfo

### ISSUE THREAD

**Issue by schlafly** [spacetelescope/rad#437]
Conceptually I had associated v2_ref, v3_ref with the v2/v3 location corresponding to radec_ref.  That's what the fields currently mean in Roman.

However, in Webb, the aberration correction is applied after v2/v3 ref, so that instead an "aberrated v2/v3" reference position is what corresponds to ra/dec_ref.  This is close to the original v2/v3 ref but scaled by va_scale.

We should consider whether we want to also provide these aberrated v2/v3 reference positions, which would make it so that we are providing the v2/v3 reference positions that actually correspond to ra/dec reference points.  Or alternatively add new ra/dec_ref that correspond to the post-aberration locations of v2/v3_ref.

**Comment by schlafly** [comment]
I think I had misunderstood this; closing.

**Comment by schlafly** [comment]
I thought I had misunderstood this but now I think my original understanding was correct; confusing!

Maybe to be more explicit about the issue: what are v2_ref, v3_ref?  I think our intent is that these are some fixed locations in the focal plane corresponding to each detector.  Following the SIAF, corresponding to the center.  The WCS contains fields v2_ref, v3_ref, ra_ref, dec_ref.  The implication is that wcs(center pixel) = ra_ref, dec_ref.  I don't think that's actually true, though, in the presence of velocity aberration.

The relevant code is here
https://github.com/spacetelescope/jwst/blob/main/jwst/assign_wcs/pointing.py#L283-L302
https://github.com/spacetelescope/jwst/blob/main/jwst/assign_wcs/nircam.py#L89-L101

The dva_corr_model scales the v2/v3 coordinates by an aberration-dependent scale factor, and then shifts all of the centers by (1 - va_scale) * v2/v3_ref.  This correction is zero only when v2/v3_ref = 0; i.e., all locations are shifted to be closer / further from the boresight.  Then tel2sky rotates the aberrated frame so that the position that has v2/v3_ref in the aberrated frame goes to ra/dec ref.  I think people would expect the original v2/v3_ref / the original center to correspond to ra/dec_ref, and I don't think that's what we're doing.  And I'm having trouble thinking about what we want to tell ~SDP to populate ra/dec_ref with.

@tddesjardins , have you thought about this?  AFAICT we're doing the same thing as Webb, but do you know what Webb populates ra/dec_ref with?

**Comment by nden** [comment]
JWST works exactly the same way. Instead of shifting ra/dec_ref we are shifting the underlying image so effectively re_ref, dec_ref project into the aberration corrected v2_ref, v3_ref. I believe the aberration parameters are recorded. We could explicitly record the the corrected V2, V3 after they are computed but it's a bit of a duplication. Is there a use case for it?

**Comment by schlafly** [comment]
Yeah, I just am not very good at reading the velocity aberration code.  This is okay and I'll close the issue again.

Trying to record more information for posterity, the realization that lets me close this again is that dva(v2ref, v3ref) == (v2ref, v3ref) exactly.  The dva correction leaves the detectors centered at their original v2/v3 reference locations and changes only their scaling.  The shift and scaling should be seen as scale * (v2 - v2_ref) + v2_ref, scale * (v3 - v3_ref) + v3_ref, even though it's written as scale * v2 + (1 - scale) * v2_ref, etc..  These are of course the same thing but I somehow wasn't getting it before.

So the va correction doesn't move v2/v3 ref, and so this issue is wrongheaded and v2/v3_ref do correspond exactly to ra_ref, dec_ref, and everything is fine.

I had mentally seen this as shrinking down the whole v2/v3 plane, which would correspond to _only_ scaling va.  I had been mentally seeing the shifts as the shifts needed to move the center of the detector to where they should go after scaling down a single detector, but they're doing the opposite of that; they're putting each detector back in its nominal v2/v3 reference location (for the central pixel), after uniformly scaling them up or down (and effectively moving the nominal v2/v3 reference location as part of that scaling).  

I had worried about what to tell people to populate ra_ref / dec_ref with (it's a weird thing from an intermediate frame ...) but realizing this now that it's exactly the location of the central pixel makes everything well defined and straightforward and explains why Jonathan's aberration code in romanisim works so well---which apparently I partially recognized a few weeks ago when I went to close this but not well enough for me to avoid opening it again.  :-(  Sorry for the noise!
